lib/internal/process/promises.js

```diff
@@ -161,13 +182,27 @@ function processPromiseRejections() {
     }
     promiseInfo.warned = true;
     const { reason, uid } = promiseInfo;
-    if (state === states.strict) {
-      fatalException(reason);
-    }
-    if (!process.emit('unhandledRejection', reason, promise) ||
-        // Always warn in case the user requested it.
-        state === states.warn) {
-      emitWarning(uid, reason);
+    switch (unhandledRejectionsMode) {
+      case kThrowUnhandledRejections: {
+        fatalException(reason);
```

Comment:
Also funny enough there really isn't anything fatal about this function, or `triggerFatalException`, as they give the user a chance to pretend nothing has happened by listening to `uncaughtException`.

Reply:
This is a legacy name. It does however reflect it's actual purpose. It is just unfortunate that we have an escape hatch for it with `process.on('uncaughtException', fn)`.